pyrpl/software_modules/lockbox/models/fabryperot.py

```diff
@@ -96,9 +96,12 @@ def fit(self, curve, voltage):
         pars = p_params + b_params
         result = mod.fit(curve, pars, x=x_for_fit, method="least_squares", 
                          max_nfev=self.max_nfev)
-
-        maximum = result.values["height"] + result.values["c"]
-        minimum = result.values["c"]
+        if not negative:
+            maximum = result.values["height"] + result.values["c"]
+            minimum = result.values["c"]
+        else:
+            maximum = result.values["c"]
+            minimum = result.values["c"] + result.values["height"]
```

Comment:
first time I get to use the XOR operator in python - maybe too much of a stretch?
```suggestion
        minimum = result.values["c"]
        maximum = minimum + result.values["height"]
        if self.negative:
            minimum, maximum = maximum, minimum
        if minimum > maximum:
            raise FitError("The fitted peak has the wrong sign.")
```